### ISSUE TITLE
Brainstorm databricks questions for Q&A session

### ISSUE THREAD

**Issue by KPeen** [Energinet-DataHub/geh-aggregations#159]
As a team, we want to compile a set of questions to provide databricks the opportunity to answer them at the Q&A session which is in the end of August

AC1: Provide Martin with a list of questions

**Comment by mknic** [comment]
1. How to be able to Query delta lake using SQL queries? Do we need to enable something to get it to work?

**Comment by kristianschneider** [comment]
How can we examine the efficiency of our queries?
Ie: where do our queries spend time? Perhaps something similar execution plans in SQL server

**Comment by kristianschneider** [comment]
Do we do our calculations the most efficient way by running them in a combined job or should we split it out into smaller jobs ?

**Comment by kristianschneider** [comment]
Is our strategy with triggering a job externally the best approach?

**Comment by kristianschneider** [comment]
Where is the most vibrant community? 
Where can we get the best help from piers?

**Comment by mknic** [comment]
Summary of above comments:

1) How do we enable feature to let us query Delta Lake using SQL queries in Databricks (Use Case: Generate reports for BI usage etc.)

2) Are there any tooling in Databricks to monitor the execution of a job to trace where queries do not perform etc.? Something similar to Execution plans on a SQL Server.

3) Do we do our calculations the most effecient way by running them step by step in a combined job? Should we split the calculations out in smaller jobs? If you decide to split out in smaller jobs how do you then share input data between jobs?

4) Is our strategy with triggering a job externally the best approach? (Current solution is to trigger a databricks job using an hhtp triggerend azure function) 

5) Where do we find the most vibrant community and thereby get the best help from piers?

**Comment by BjarkeMeier** [comment]
Contribution from team Volt:

## Developer Experience

Out development environment feels complex and brittle. A review of the concepts with containers, VSCode, launch and debugging settings and more would be nice.

## Streaming

The streaming job is located [here](https://github.com/Energinet-DataHub/geh-timeseries/blob/main/source/streaming/enrichment_and_validation.py).

It would be nice to have a discussion about whether the following things seems reasonable or what can be improved.

* How do we get confident that the streaming is running fine? How do we prepare for eventual failure?
* What are the alternatives to the concept for re-fetching new master data every 5 minutes (see the end of the streaming job)?
  Ideally we would like to have a streaming source of changes to master data.
* We could use a hand to eliminate all warnings and errors from executing the streaming job
* We have a number of TODOs in the code that we would like some input on
* ProtoBuf: Are there known plans about support? How does the (work in progress) approach in [this branch](https://github.com/Energinet-DataHub/geh-timeseries/blob/Use_protobuf_for_received_time_series/source/streaming/geh_stream/streaming_utils/input_source_readers/time_series_reader.py) look?
* How do we handle transactions in streaming batches where we both store data in Delta Lake and publish events?

## Testing

* Are there better/simpler ways to write unit tests?
  Perhaps have a look at the tests [here](https://github.com/Energinet-DataHub/geh-timeseries/blob/main/source/streaming/tests/streaming_utils/streamhandlers/test_enricher.py)
  including the concept of a fixture factory - `parsed_data_factory` from [conftest.py](https://github.com/Energinet-DataHub/geh-timeseries/blob/main/source/streaming/tests/conftest.py).

## Deployment

Is building and deploying a wheel the best option?

## PySpark

* Are there any caveats regarding date and time zones?

**Comment by kristianschneider** [comment]
How does DBR play together with CosmosDB (many nodes talking with cosmos at the same time)

**Comment by kristianschneider** [comment]
Would it be possible to run storing dataframes in parallel while running calculations ( Snapshot)
